src/main/scala/io/treeverse/clients/LakeFSContext.scala

```diff
@@ -45,4 +48,31 @@ object LakeFSContext {
       classOf[WithIdentifier[Catalog.Entry]],
     )
   }
+
+  def newDF(
+      spark: SparkSession,
+      repoName: String,
+      commitID: String,
+  ): DataFrame = {
+    val rdd = newRDD(spark.sparkContext, repoName, commitID).map { pair =>
+      val key = pair._1
+      val entry = pair._2.message
```

Comment:
```suggestion
    val rdd = newRDD(spark.sparkContext, repoName, commitID).map { case (key, entry) =>
```
(I think)

E.g. this works:
```scala
scala> def l = Seq((1,2), (3,4), (5,6))
def l: Seq[(Int, Int)]

scala> l
val res0: Seq[(Int, Int)] = List((1,2), (3,4), (5,6))

scala> l.map({ case (a,b) => a+b })
val res1: Seq[Int] = List(3, 7, 11)
```